lib/httpDigest.js

```diff
@@ -19,13 +22,64 @@ async function create({data, algorithm = 'sha256', useMultihash = true}) {
   if(typeof data !== 'string') {
     data = JSON.stringify(data);
   }
+  algorithm = algorithm.toLowerCase();
+  const digest = await _getDigest({data, algorithm});
+  if(useMultihash) {
+    return _createMultiHash({digest});
+  }
+  return `SHA-256=${base64Encode(digest)}`;
+}
+
+/**
+ * Verifies digest of data created by create().
+ *
+ * @param {object} options - The options to use.
+ * @param {string|object} options.data - The data to be verified.
+ * @param {string} options.header - The digest header string to
+ *   be verified against.
+ *
+ * @returns {object} - Returns an object with verified true or false.
+ */
+async function verify({data, header}) {
+  const {algorithm, value: headerValue} = _parseHashedString(header);
+  let hashedDigestString;
+  let hashedDigestStringValue;
```

Comment:
These variable names are hard to follow... and I think they are probably the result of not breaking this up into its logical pieces quite right. I think this is too hard to make suggestions for, so I'll push a commit to take a look at.